phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -584,16 +633,19 @@ public boolean next(List<Cell> results) throws IOException {
             if (onlyVerify) {
                 addToBeVerifiedIndexRows();
             }
-            ArrayList<KeyRange> keys = new ArrayList<>(rowCountPerTask);
+            ArrayList<KeyRange> keyRanges = new ArrayList<>(rowCountPerTask);
+            Set<byte[]> indexRowKeys = new HashSet<>(rowCountPerTask);
```

Comment:
A byte array hash set would not work here. The byte array needs to be wrapped with equals and hashCode, or Map<byte[], Put> created with Maps.newTreeMap(Bytes.BYTES_COMPARATOR) can be used instead.